File: 00. Estructura/01. Lenguajes de proposito general/04. Paradigmas de programación/02. Orientada a objetos/01. Clases y objetos/Python/Clases y objetos.py

# Creación de la clase
class Clase:
    
    # Atributos
    atributo = valor

    # Metodo constructor (Solo sobrevive el ultimo constructor definido)
    def __init__(self, param1, param2, paramN):
        # Contenido
        ''' Contexto ''' self.atributo1 # Contexto                   # Referencia a atributo
        ''' Contexto ''' self.metodo(val1, val2, val3) # Contexto    # Referencia a metodo
        # Contenido

    # Metodos
    def metodo(self, param1, param2, paramN):
        # Contenido

# Creación e instanciación
objeto = Clase(atributo1, atributo2, atributoN)     # Creación e instanciación

# Llamado 
''' Contexto ''' objeto # Contexto                             # Llamado al objeto
''' Contexto ''' objeto.metodo(val1, val2, valN) # Contexto    # Llamado a un metodo
''' Contexto ''' objeto.atributo # Contexto                    # Llamado a un atributo

# Eliminación
del objeto      # Eliminar objeto 
del Clase       # Eliminar clase 

File: 01. Lenguajes de proposito general/04. Paradigmas/02. Orientada a objetos/01. Clases y objetos/JavaScript/Clases y objetos.js

// Creación de la clase
class Clase {
    
    // Atributos
    atributo = valor;

    // Metodo constructor (Solo puede haber uno)
    constructor(param1, param2, paramN) {
        // Contenido
        /* Contexto */ this.atributo /* Contexto */                  // Referencia a atributo
        /* Contexto */ this.metodo(val1, val2, val3) /* Contexto */  // Referencia a metodo
        // Contenido
    }

     // Metodos
    metodo(param1, param2, paramN) {
        // Contenido
    }
    
}

// Creación e instanciación
objeto = new Clase(val1, val2, valN); // Creación e instanciación

// Llamado 
/* Contexto */ objeto /* Contexto */                            // Llamado al objeto
/* Contexto */ objeto.metodo(val1, val2, valN) /* Contexto */   // Llamado a un metodo
/* Contexto */ objeto.atributo /* Contexto */                   // Llamado a un atributo

// Eliminación
NO SOPORTADO (EL GARBAGE COLLECTOR ELIMINA ELEMENTOS AUTOMATICAMENTE CUANDO YA SON INACCESIBLES)
